packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,29 @@
 import Objection from 'objection';
+import {ChannelResult} from '@statechannels/client-api-schema';
 
 import {SignState} from '../protocols/actions';
 import {Channel, SyncState} from '../models/channel';
+import {Bytes32} from '../type-aliases';
+import {ChannelState} from '../protocols/state';
 
 export const Store = {
-  signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
+  signState: async function(
+    action: SignState,
+    tx: Objection.Transaction
+  ): Promise<{outgoing: SyncState; channelResult: ChannelResult}> {
     const channel = await Channel.forId(action.channelId, tx);
-    const notifications = channel.signAndAdd(action);
+    const outgoing = channel.signAndAdd(action);
     await Channel.query(tx).update(channel);
 
-    return notifications;
+    const {channelResult} = channel;
+
+    return {outgoing, channelResult};
+  },
+
+  getChannel: async function(
+    channelId: Bytes32,
+    tx: Objection.Transaction | undefined
+  ): Promise<ChannelState> {
```

Comment:
This could be an `Option`

Reply:
```ts
const dbResponse = await Channel.forId(channelId, tx);
if (!dbResponse) return none;
return some(dbResponse.protocolState);
```

Reply:
I am not really convinced that `Option` is worth the overhead, though you did point out a bug in the code: the return type (without option) should be
A. `Promise<ChannelState | undefined>`; or
B. `Promise<Either<ChannelNotFoundError, ChannelState>>`

Reply:
Resolved in 572a3ec. I opted to return `ChannelState | undefined` for now.